core/tx_pool.go

```diff
@@ -518,34 +518,34 @@ func (pool *TxPool) validateTx(tx *types.Transaction, local bool) error {
 	}
 	// Ensure the transaction doesn't exceed the current block limit gas.
 	if pool.currentMaxGas < tx.Gas() {
-		return ErrGasLimit
+		return fmt.Errorf("%v currentMaxGas:%d;tx.Gas():%d", ErrGasLimit, pool.currentMaxGas, tx.Gas())
 	}
 	// Make sure the transaction is signed properly
 	from, err := types.Sender(pool.signer, tx)
 	if err != nil {
-		return ErrInvalidSender
+		return fmt.Errorf("%v err is:%v", ErrInvalidSender, err)
 	}
 	// Drop non-local transactions under our own minimal accepted gas price
 	local = local || pool.locals.contains(from) // account may be local even if the transaction arrived from the network
 	if !local && pool.gasPrice.Cmp(tx.GasPrice()) > 0 {
-		return ErrUnderpriced
+		return fmt.Errorf("%v pool.gasPrice:%d;tx.GasPrice():%d", ErrUnderpriced, pool.gasPrice, tx.GasPrice())
 	}
 	// Ensure the transaction adheres to nonce ordering
 	if pool.currentState.GetNonce(from) > tx.Nonce() {
-		return ErrNonceTooLow
+		return fmt.Errorf("%v current pool nonce:%d;tx.Nonce():%d", ErrNonceTooLow, pool.currentState.GetNonce(from), tx.Nonce())
 	}
 	// Transactor should have enough funds to cover the costs
 	// cost == V + GP * GL
 	if pool.currentState.GetBalance(from).Cmp(tx.Cost()) < 0 {
-		return ErrInsufficientFunds
+		return fmt.Errorf("%v your balance:%d;tx.Cost():%d", ErrInsufficientFunds, pool.currentState.GetBalance(from), tx.Cost())
```

Comment:
1. `ErrInsufficientFunds `is also used in unit test

2. `pool.currentState.GetBalance(from)` and `tx.Cost()` are both big.Int type. It should be printed via `%v` instead of `%d`